### ISSUE TITLE
Amazon Cloud Drive certificate error

### ISSUE THREAD

**Issue by zaggynl** [OpenMediaVault-Plugin-Developers/openmediavault-duplicati#1]
Hi, thanks for bringing Duplicati to OMV!
Am trying to backup to Amazon Cloud Drive.
When pressing Test Connection I receive the following error:

Failed to connect: The server certificate had the error RemoteCertificateChainErrors and the hash C7DCE9F5333DE5E8FC35F2FCC411E5C0909B032F If you trust this certificate, use the commandline option --accept-specified-ssl-hash=C7DCE9F5333DE5E8FC35F2FCC411E5C0909B032F to accept the server certificate anyway. You can also attempt to import the server certificate into your operating systems trust pool. You may want to import a set of trusted certificates into the Mono certificate store. Use the command: cert-sync /etc/ssl/certs/ca-certificates.crt #for Debian based systems cert-sync /etc/pki/tls/certs/ca-bundle.crt #for RedHat derivatives Read more: http://www.mono-project.com/docs/about-mono/releases/3.12.0/#cert-sync 

Need to run cert-sync, not installed?
Installed mono
sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
echo "deb http://download.mono-project.com/repo/debian wheezy main" | sudo tee /etc/apt/sources.list.d/mono-xamarin.list
sudo apt-get update && sudo apt-get install mono-complete
cert-sync /etc/ssl/certs/ca-certificates.crt
I already trust 174, your new list has 174
So appears already up to date.

Am not sure how to proceed, blind accept cert with --accept-specified-ssl-hash=6942386283C265EFFD4C04AD809706D0800E9315 ?

**Comment by ryecoaaron** [comment]
I don't think you need to add the mono-project repo.  All of my testing has been done using the mono packages in the Debian repo.

And yes, I added the accept-specified-ssl-hash.  I was able to create a folder on my Amazon account but still couldn't get backups working.  So, I'm not the best to answer these questions.  Perhaps, the duplicati forum or github would be better.

**Comment by zaggynl** [comment]
Thanks for replying!

Yeah I still run into a cert error when actually trying to make the backup with the accept ssl hash advanced option:
Fatal error
System.Net.WebException: Error getting response stream (Write: The authentication or decryption has failed.): SendFailure ---> System.IO.IOException: The authentication or decryption has failed. ---> Mono.Security.Protocol.Tls.TlsException: Invalid certificate received from server.

Oh well, rclone it is.

**Comment by ryecoaaron** [comment]
You didn't want to ask on the duplicati forum?

**Comment by zaggynl** [comment]
Well I've tried duplicati stand alone before, got it to work in the end but it suffered from high memory use and breaks frequently, hence the experimental I guess.
This was my last ditch attempt.

**Comment by ryecoaaron** [comment]
I think cert error should be fixed in the latest plugin since it adds the correct certs.

**Comment by Vrumfondel** [comment]
This is a quite minor issue in Duplicati so far.
Using Amazon Cloud Drive is not working at all right now. There must be some glitch with Duplicati running in debian environment.
Bug is reported there allready.

And yes, the certificate problems are noted there aswell allready.
